events/log_entries.go

```diff
@@ -29,6 +31,7 @@ import (
 
 // LogEntry is the log entry representation
 type LogEntry struct {
+	ctx            context.Context
```

Comment:
Why storing context here?
Except for a very specific good reason this is a code smell: 
From https://github.com/golang/go/wiki/CodeReviewComments#contexts

> Don't add a Context member to a struct type; instead add a ctx parameter to each method on that type that needs to pass it along. The one exception is for methods whose signature must match an interface in the standard library or in a third party library.

And from https://golang.org/pkg/context/ :
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:

Reply:
Hey, this is a very specific good reason as the log entry is built with an existing context. No sense to have another context parameter in the register method

Reply:
:+1: